main/SQLHelper.cpp

```diff
@@ -5632,6 +5665,22 @@ bool CSQLHelper::GetLastValue(const int HardwareID, const char* DeviceID, const
 	return result;
 }
 
+std::string  CSQLHelper::GetDeviceValue(const char* FieldName, const std::string& Idx)
+{
+	bool result = false;
+	std::vector<std::vector<std::string> > sqlresult;
+
+    sqlresult = safe_query("SELECT %s FROM DeviceStatus WHERE ( ID= %s )" , FieldName, Idx.c_str() );
```

Comment:
please use the 'auto' keyword instead of declaring the vector, can also be a online liner

Reply:
Ok done.